plugins/catalog/src/routes.ts

```diff
@@ -24,6 +24,10 @@ export const createComponentRouteRef = createExternalRouteRef({
   optional: true,
 });
 
+/**
+ * Route ref for linking to Tech docs page
+ * @public (undocumented)
+ */
 export const viewTechDocRouteRef = createExternalRouteRef({
```

Comment:
I tried that at first, but I couldn't make it work. the `createExternalRouteRef` only keeps returning `undefined`, even though i'm doing the exact same routing configuration as the catalog plugin.

Reply:
I've tried the following setup, which is exactly the same as in catalog-plugin:

```tsx
// routes.ts
export const viewTechDocRouteRef = createExternalRouteRef({
  id: 'view-techdoc',
  optional: true,
  params: ['namespace', 'kind', 'name'],
});
```

```tsx
// plugin.ts
import { viewTechDocRouteRef } from './routes';

externalRoutes: {
  viewTechDoc: viewTechDocRouteRef,
},
```

```tsx
// TemplateCard.tsx
import { viewTechDocRouteRef } from '../../routes';

const viewTechDoc = useRouteRef(viewTechDocRouteRef);
const viewTechDocsAnnotation =
  template.metadata.annotations?.['backstage.io/techdocs-ref'];
const viewTechDocsLink =
  !!viewTechDocsAnnotation &&
  !!viewTechDoc &&
  viewTechDoc({
    namespace: template.metadata.namespace || DEFAULT_NAMESPACE,
    kind: template.kind,
    name: template.metadata.name,
  });

console.log(viewTechDoc).       // `undefined`
```

Are you familiar with the external routing mechanism, and if so, can you help me a bit? 😇

Reply:
So it should export it's own, and then it will need to be bound to the right place in your own app.

Something like this: https://github.com/backstage/backstage/blob/d3737da3377da376b0b82c9d884cc656f5455292/packages/app/src/App.tsx#L128-L131

Reply:
As for the `next` template cards, they live here: https://github.com/backstage/backstage/blob/3400190663cab1bab52f52bf15d0c967d42517b0/plugins/scaffolder/src/next/TemplateListPage/TemplateCard/TemplateCard.tsx#L63
 
And you should be able to visit them in the main repo, by going to `localhost:7007/create/next` and that should show all the work on the next branch.

Reply:
I've cleaned this up 😄